packetWin7/npf/npf/Packet.c

```diff
@@ -351,6 +374,14 @@ DriverEntry(
 	RtlInitUnicodeString(&strKeGetCurrentProcessorNumberEx, L"KeGetCurrentProcessorNumberEx");
 	g_My_KeGetCurrentProcessorNumberEx = (KEGETCURRENTPROCESSORNUMBEREX) NdisGetRoutineAddress(&strKeGetCurrentProcessorNumberEx);
 
+	//

```

Comment:
If the test for Windows 8 and later is removed, will this still work?  Presumably `MmGetSystemRoutineAddress()` will return NULL if `KeQuerySystemTimePrecise()` isn't available, so `My_KeQuerySystemTime()` will fall back on `KeQuerySystemTime()`.

Reply:
You are right, the test is redundant, I will remove it. I inserted it before reading that `MmGetSystemRoutineAddress()` returns NULL on failure.